microcosm/opaque.py

```diff
@@ -39,9 +44,52 @@ def __enter__(self):
             self.saved = deepcopy(opaque._store)
             opaque.update(self.func())
 
+            # Use of a tracing solution on top of opaque is optional!
+            if opaque.tracer:
+                span = self.start_span()
```

Comment:
These three functions are important and amount to the collective learning I did about the `jaeger` API.

First we start a new span trying to continue an existing trace referenced in `graph.opaque` otherwise starting a span in an effectively new trace.

Then we copy tags from `graph.opaque` into the current span.

Finally we add the information about the current trace to `graph.opaque`. This way other `microcosm-` libraries don't know about `jaeger` at all and we can swap it out for other tracing software by only updating microcosm.